java/core/lancedb-jni/src/ffi.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright 2024 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::slice;
+
+use jni::objects::{JByteBuffer, JObjectArray, JString};
+use jni::sys::jobjectArray;
+use jni::{objects::JObject, JNIEnv};
+
+use crate::error::{Error, Result};
+
+/// TODO(lu) import from lance-jni without duplicate
+/// Extend JNIEnv with helper functions.
+pub trait JNIEnvExt {
```

Comment:
will we add unit tests?

Reply:
Yeah, i added some examples of test cases but haven't got time to add more. 
https://github.com/lancedb/lance/blob/main/java/core/src/test/java/com/lancedb/lance/JNITest.java

Because testing JNIEnv requires the launch of an actual JVM, so i plan to add the test on the Java side.
Ideally the tests will be passing some object from Java to Rust, Rust side consumes the Java object, convert to rust object, and convert back to java object to return to the Java side. If anything is wrong, the test will fail